dbt/adapters/snowflake/relation_configs/dynamic_table.py

```diff
@@ -21,6 +35,147 @@ class SnowflakeDynamicTableConfig(RelationConfigBase):
+            kwargs_dict.update(
+                {"schema_name": cls._render_part(ComponentName.Schema, schema_name)}
+            )
+        if database_name := config_dict.get("database_name"):
+            kwargs_dict.update(
+                {"database_name": cls._render_part(ComponentName.Database, database_name)}
+            )
+
+        if target_lag := config_dict.get("target_lag"):
+            kwargs_dict.update(
+                {"target_lag": SnowflakeDynamicTableTargetLagConfig.from_dict(target_lag)}
+            )
+
+        dynamic_table: "SnowflakeDynamicTableConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return dynamic_table
+
+    @classmethod
+    def parse_model_node(cls, model_node: ModelNode) -> dict:
+        config_dict = {
+            "name": model_node.identifier,
+            "schema_name": model_node.schema,
+            "database_name": model_node.database,
+            "query": model_node.compiled_code,
+            "target_lag": SnowflakeDynamicTableTargetLagConfig.parse_model_node(model_node),
+            "warehouse": model_node.config.extra.get("warehouse"),
+        }
+        return config_dict
+
+    @classmethod
+    def parse_relation_results(cls, relation_results: RelationResults) -> dict:
+        if dynamic_table := relation_results.get("dynamic_table"):
+            dynamic_table_config = dynamic_table.rows[0]
+        else:
+            dynamic_table_config = {}
+
+        config_dict = {
+            "name": dynamic_table_config.get("name"),
+            "schema_name": dynamic_table_config.get("schema_name"),
+            "database_name": dynamic_table_config.get("database_name"),
+            "query": cls._parse_query(dynamic_table_config.get("text")),
+            "target_lag": SnowflakeDynamicTableTargetLagConfig.parse_relation_results(
+                relation_results
+            ),
+            "warehouse": dynamic_table_config.get("warehouse"),
+        }
+        return config_dict
+
+    @classmethod
+    def _parse_query(cls, query: str) -> str:
+        """
+        Get the select statement from the dynamic table definition in Snowflake.
+
+        Args:
+            query: the `create dynamic table` statement from `show dynamic tables`, for example:
+
+            create dynamic table my_dynamic_table
+                target_lag = '1 minute'
+                warehouse = MY_WAREHOUSE
+                as (
+                    select * from my_base_table
+                )
+            ;
+
+        Returns: the `select ...` statement, for example:
+
+            select * from my_base_table
+
+        """
+        open_paren = query.find("as (") + len("as (")
+        close_paren = query.rindex(")")
+        return query[open_paren:close_paren].strip()
+
+
+@dataclass(frozen=True, eq=True, unsafe_hash=True)
+class SnowflakeDynamicTableWarehouseConfigChange(
+    RelationConfigChange, RelationConfigValidationMixin
+):
+    context: str
```

Comment:
Where does this get used?

Reply:
It's used in the alter statement in the jinja here: https://github.com/dbt-labs/dbt-snowflake/blob/f9e80b679bf08360b59ee2c0349caeef465d7157/dbt/include/snowflake/macros/materializations/dynamic_table/ddl.sql#L24